src/coreclr/jit/rationalize.cpp

```diff
@@ -265,6 +265,67 @@ void Rationalizer::RewriteIntrinsicAsUserCall(GenTree** use, ArrayStack<GenTree*
                       arg1, arg2);
 }
 
+#ifdef TARGET_ARM64
+// RewriteSubLshDiv: Possibly rewrite a SubLshDiv node into a Mod.
+//
+// Arguments:
+//    use - A use of a node.
+//
+// Transform: a - (a / cns) >> shift  =>  a % cns1
+//            where cns is an signed integer constant that is a power of 2.
+// We do this transformation because Lowering has a specific optimization
+// for 'a % cns1' that is not easily reduced by other means.
+//
+void Rationalizer::RewriteSubLshDiv(GenTree** use)
+{
+    GenTree* const node = *use;
+
+    if (!node->OperIs(GT_SUB))
+        return;
+
+    GenTree* op1 = node->gtGetOp1();
+    GenTree* op2 = node->gtGetOp2();
+
+    if (!(node->TypeIs(TYP_INT, TYP_LONG) && !node->IsUnsigned() && op1->OperIs(GT_LCL_VAR)))
```

Comment:
```suggestion
    if (!(node->TypeIs(TYP_INT, TYP_LONG) && op1->OperIs(GT_LCL_VAR)))
```

This is always true, `GT_SUB` cannot be unsigned.

Reply:
Good catch, yeah it should rather be changed to `!node->gtOverflow()`.

Reply:
Does overflow matter here? When this is fully transformed, the divisor has to be a constant power of 2 which could never overflow.

Reply:
Does overflow matter here? When this is fully transformed, the divisor has to be a constant power of 2 which could never overflow.

Reply:
No, this probably cannot overflow. But in that case I still think the `IsUnsigned()` check does not make much sense.

Reply:
If `GT_SUB` is guaranteed to never be `unsigned` then I guess we can remove the check.